custom_components/myskoda/binary_sensor.py

```diff
@@ -317,6 +317,32 @@ def is_on(self):  # noqa: D102
         return self.vehicle.status.detail.bonnet == OpenState.OPEN
 
 
+class SunroofOpen(MySkodaBinarySensor):
+    """Detectes whether the sunroof is open."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, vehicle: Vehicle) -> None:  # noqa: D107
+        super().__init__(
+            coordinator,
+            vehicle,
+            BinarySensorEntityDescription(
+                key="sunroof_open",
+                name=f"{vehicle.info.specification.title} Sunroof",
+                device_class=BinarySensorDeviceClass.OPENING,
+                icon="mdi:car-select",
+            ),
+        )
+        self._attr_unique_id = f"{vehicle.info.vin}_sunroof_open"
+
+    @property
+    def is_on(self):  # noqa: D102
+        if not self.coordinator.data:
+            return None
+
+        self._update_device_from_coordinator()
+
+        return self.vehicle.status.detail.sunroof == OpenState.OPEN
```

Comment:
This way, when a vehicle doesn't have a sunroof, it will be reported as "CLOSED". I'd suggest returning `None` in that case for now.

We should find a way to leverage capabilities in the future, in order to not enable sensors that don't make sense.

Reply:
I'll create an issue for that so we dont forget, and patch this accordingly for now